test/unit/math/prim/prob/hmm_marginal_test.cpp

```diff
@@ -0,0 +1,243 @@
+#include <stan/math/prim/prob/hmm_marginal_lpdf.hpp>
+#include <boost/math/distributions.hpp>
+#include <boost/random.hpp>
+#include <test/unit/math/test_ad.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+
+/**
+ * Wrapper around hmm_marginal_density which passes rho and
+ * Gamma without the last element of each column. We recover
+ * the last element using the fact each column sums to 1.
+ * The purpose of this function is to do finite diff benchmarking,
+ * without breaking the simplex constraint.
+ */
+template <typename T_omega, typename T_Gamma, typename T_rho>
+inline stan::return_type_t<T_omega, T_Gamma, T_rho> hmm_marginal_test_wrapper(
+    const Eigen::Matrix<T_omega, Eigen::Dynamic, Eigen::Dynamic>& log_omegas,
+    const Eigen::Matrix<T_Gamma, Eigen::Dynamic, Eigen::Dynamic>&
+        Gamma_unconstrained,
+    const std::vector<T_rho>& rho_unconstrained) {
+  using stan::math::row;
+  using stan::math::sum;
+  int n_states = log_omegas.rows();
+
+  Eigen::Matrix<T_Gamma, Eigen::Dynamic, Eigen::Dynamic> Gamma(n_states,
+                                                               n_states);
+  for (int i = 0; i < n_states; i++) {
+    Gamma(i, n_states - 1) = 1 - sum(row(Gamma_unconstrained, i + 1));
+    for (int j = 0; j < n_states - 1; j++) {
+      Gamma(i, j) = Gamma_unconstrained(i, j);
+    }
+  }
+
+  Eigen::Matrix<T_rho, Eigen::Dynamic, 1> rho(n_states);
+  rho(1) = 1 - sum(rho_unconstrained);
+  for (int i = 0; i < n_states - 1; i++)
+    rho(i) = rho_unconstrained[i];
+
+  return stan::math::hmm_marginal_lpdf(log_omegas, Gamma, rho);
+}
+
+/**
+ * In the proposed example, the latent sate x determines
+ * the observational distribution:
+ *  (i) normal(mu, sigma)
+ *  (ii) normal(-mu, sigma)
+ */
+double state_lpdf(double y, double abs_mu, double sigma, int state) {
+  int x = (-2 * state + 1);
+  double chi = (y - x * abs_mu) / sigma;
+  return -0.5 * chi * chi - 0.5 * std::log(6.283185307179586) - std::log(sigma);
```

Comment:
Please rewrite so as not to make it an algebra puzzle for the reader:

```
return stan::math::normal_lpdf<false>(y, state == 0 ? -mu : mu, sigma);
```

Seeing `2 * pi` in decimal form is a strong clue something's fishy.

Reply:
How is this
```
double state_lpdf(double y, double abs_mu, double sigma, int state) {
  int x = state == 0 ? 1 : -1;
  double chi = (y - x * abs_mu) / sigma;
  return -0.5 * chi * chi - 0.5 * std::log(2 * M_PI) - std::log(sigma);
}
```
?